tools/ansible/tasks/wsu/main.yaml

```diff
@@ -20,25 +20,151 @@
+
+    - name: Get kubernetes minor version
+      shell: "oc version -o json | jq -r '.serverVersion.minor' | tr -d +"
+      register: kubernetes_minor_version
+
+    # This translates a kubernetes minor version to an OpenShift version. This works under the assumption that each
+    # OpenShift release corresponds with a kubernetes release. This is being done this way, and not with oc get
+    # clusterversion, as OpenShift CI doesn't have the actual version attached to its clusters, instead replacing it
+    # with 0.0.1 and information about the release creation date
+    - name: Get cluster version
+      shell: |
+        MINORVERSION={{ kubernetes_minor_version.stdout }}
+        BASEK8SVERSION=16
+        BASEOPENSHIFTVERSION=3
+        if [ \"$MINORVERSION\" -lt \"$BASEK8SVERSION\" ]; then
+        echo unsupported kubernetes version; exit 1
+        fi
+        VERSIONINCREMENTS=$((MINORVERSION - BASEK8SVERSION))
+        echo 4.$((BASEOPENSHIFTVERSION + VERSIONINCREMENTS))
+        exit 0
+      register: cluster_version
+
+    # curl -s https://api.github.com/repos/openshift/windows-machine-config-bootstrapper/releases gets the list of all
+    # releases, we use the cluster version from previous ansible task to select the release we need.
+    - name: Get release
+      shell: |
+        RELEASES="$(curl -s https://api.github.com/repos/openshift/windows-machine-config-bootstrapper/releases)"
+        TAGNAME="$(echo $RELEASES | jq '.[]' | jq '.tag_name' | \
+        grep -i {{ cluster_version.stdout }} |\
+        sort -r | awk 'NR==1')"
+        echo $RELEASES | \
+        jq '.[] | select (.tag_name == '$TAGNAME')'
+      register: release
+
+    - name: Check release object
+      fail:
+        msg: "Cannot find the latest release"
+      when: release is not defined or release.stdout == ""
+
     - name: Gather required files
       block:
         - name: Create directory to store files to be used later
           tempfile:
             state: directory
           register: tmp_dir
 
+        # Executed only if 'build_wmcb' is set to True. Default option is downloading the latest release of WMCB
+        # depending on the cluster version
         - name: Build WMCB
-          make:
-            target: build
-            chdir: "{{ project_root }}"
-
-        - name: Move WMCB to temporary directory
-          command: mv "{{ project_root }}/{{ wmcb_exe }}" "{{ tmp_dir.path }}/{{ wmcb_exe }}"
-
+          # 'build_wmcb' is used as a boolean - it has to be provided in JSON format
+          # https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#passing-variables-on-the-command-line
+          when: build_wmcb is defined and build_wmcb
+          block:
+          - name: Build WMCB
+            make:
+              target: build
+              chdir: "{{ project_root }}"
+
+          - name: Move WMCB to temporary directory
```

Comment:
We should build WMCB directly in the temp directory. Thats a solution to the bug @akhil-rane is reporting.